kylin/src/main/java/org/apache/zeppelin/kylin/KylinInterpreter.java

```diff
@@ -0,0 +1,197 @@
+            .add(KYLIN_QUERY_PROJECT, "default", "kylin paroject name")
+            .add(KYLIN_QUERY_OFFSET, "0", "kylin query offset")
+            .add(KYLIN_QUERY_LIMIT, "5000", "kylin query limit")
+            .add(KYLIN_QUERY_ACCEPT_PARTIAL, "true", "The kylin query partial flag").build());
+  }
+
+
+
+
+  public KylinInterpreter(Properties property) {
+    super(property);
+  }
+  @Override
+  public void open() {
+
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context) {
+    try {
+      return executeQuery(st);
+    } catch (IOException e) {
+      logger.info("failed to query data in kylin ", e.getMessage());
+      return new InterpreterResult(InterpreterResult.Code.ERROR, e.getMessage());
+    }
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) {
+
+  }
+
+  @Override
+  public FormType getFormType() {
+    return FormType.SIMPLE;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) {
+    return 0;
+  }
+
+  @Override
+  public List<String> completion(String buf, int cursor) {
+    return null;
+  }
+
+  private InterpreterResult executeQuery(String sql) throws IOException {
+
+    logger.info("project" + getProperty(KYLIN_QUERY_PROJECT));
+    logger.info("sql:" + sql);
+    logger.info("acceptPartial:" + getProperty(KYLIN_QUERY_ACCEPT_PARTIAL));
+    logger.info("limit:" + getProperty(KYLIN_QUERY_LIMIT));
+    logger.info("offset:" + getProperty(KYLIN_QUERY_OFFSET));
+    StringBuilder msg = null;
+    byte[] encodeBytes = Base64.encodeBase64(new String(getProperty(KYLIN_USERNAME)
+        + ":" + getProperty(KYLIN_PASSWORD)).getBytes());
+    HttpClient httpClient = HttpClientBuilder.create().build();
+
+    String postContent = new String("{\"project\":" + "\"" + getProperty(KYLIN_QUERY_PROJECT) + "\""
+        + "," + "\"sql\":" + "\"" + sql + "\""
+        + "," + "\"acceptPartial\":" + "\"" + getProperty(KYLIN_QUERY_ACCEPT_PARTIAL) + "\""
+        + "," + "\"offset\":" + "\"" + getProperty(KYLIN_QUERY_OFFSET) + "\""
+        + "," + "\"limit\":" + "\"" + getProperty(KYLIN_QUERY_LIMIT) + "\"" + "}");
+    logger.info("post:" + postContent);
+    postContent = postContent.replaceAll("[\u0000-\u001f]", " ");
+    StringEntity entity = new StringEntity(postContent, "UTF-8");
+    entity.setContentType("application/json; charset=UTF-8");
+
+    logger.info("post url:" + getProperty(KYLIN_QUERY_API_URL));
+
+    HttpPost postRequest = new HttpPost(getProperty(KYLIN_QUERY_API_URL));
+    postRequest.setEntity(entity);
+    postRequest.addHeader("Authorization", "Basic " + new String(encodeBytes));
+    postRequest.addHeader("Accept-Encoding", "UTF-8");
+    HttpResponse response = httpClient.execute(postRequest);
+
+    if (response.getStatusLine().getStatusCode() != 200) {
+      logger.info("failed", response.getEntity().getContent().toString());
```

Comment:
1. The log message `failed` is too short. 
2. The content won't be printed at all because the message doesn't use a placeholder for it, i.e. `{}`